content/blog/open-source-monitoring-at-the-hpe-customer-technology-center-böblingen.md

```diff
@@ -0,0 +1,377 @@
+---
+title: Open Source Monitoring at the HPE Customer Technology Center Böblingen
+date: 2024-01-02T15:34:10.449Z
+author: "Thomas Beha "
+authorimage: /img/tb10092023.jpg
+thumbnailimage: /img/ctcracktemppower.small.png
+disable: false
+tags:
+  - ILO
+  - Redfish
+  - Prometheus
+  - Grafana
+  - ilo-restful-api
+---
+# The Challenge
+
+T﻿he HPE Customer Technology Center (CTC ) Böblingen is a customer visit center in Germany where a wide range of HPE solutions are demonstrated. The core infrastructure of the CTC, based on HPE SimpliVity, is monitored (perfomance, capacity and uptime) already for some time using Prometheus and Grafana. The implementation of the SimpliVity Prometheus connector was described in the [HPE SimpliVity Prometheus connector whitepaper.](https://www.hpe.com/psnow/doc/a50000514enw)  This monitoring environment was extended over time with the [HPE array exporter](https://github.com/hpe-storage/array-exporter) for HPE storage arrays (HPE Alletra 6000, HPE Alletra 9000 and HPE Alletra MP).
+
+In the last summers we did face the challenge, that the initially planned cooling capacity was not sufficient anymore for the new generations of server and storage that we need to deploy in the CTC. As a consequence the lab temperature was rising to a level, that caused several incidents, where HPE ProLiant servers were automatically shutting down to avoid damages. In order to avoid this situation we were looking for a CTC lab temperature warning system that is easy to implement. An heads up of the rising temperature in the CTC would allow us to shutdown equipment that is currently not urgently needed for customer demos - helping us to ensure that the booked customer demos could still run.
+
+# Redfish® iLO RESTful API as a solution?
+
+Each HPE ProLiant server has a lot of temperature sensors spread across the system board, CPU, memory and power supplies. This detailed temperature data is available and visualized on each ILO interface in the Power&Thermal section:
+
+![](/img/ilotemperature.png "ILO Temperature Information")
+
+The first sensor, 01-Inlet Ambient temperature, is measuring the inlet temperature at the front of the server and can be used as an indicator of the CTC temperature.  Now that I have identified a usable data point an automated reading of the temperature values is needed. Having used the SimpliVity REST API to build the SimpliVity-Prometheus connector, that is used to monitor the core infrastructure, it was a natural step for me to look into the [Redfish® iLO RESTful API](https://developer.hpe.com/platform/ilo-restful-api/home/). The Redfish® API ecosystem is an open industry-standard specification (published by the Distributed Management Task Force [DTMF](http://www.dmtf.org/standards/redfish)) and provides remote server provisioning, inventory and monitoring.
```

Comment:
When you say "can be used as an indicator of the CTC temperature" do you mean "the temperature of the CTC infrastructure environment overall"? If so, you might want to clarify that. Easily done with:
"can be used as an indicator of the temperature of the overall CTC environment."

In the second sentence, place a comma after "a usable data point" so it reads "...a usable data point, an automated..."

Should SimpliVity have HPE in front of it? Usually Brand tends to be picky about ensuring that HPE is in front of every product name.

Ah, here I see you use "SimpliVity-Prometheus". That works. I'd recommend using that notation similarly above. Like I said, it felt like it needed a slash or something. A dash works just fine. Just be consistent on whichever one you choose.

In the 3rd sentence, modify to:
"Since I had already used the HPE SimpliVity REST API to build the HPE SimpliVity-Prometheus connector used to monitor the core infrastructure, it was only natural that I would look into the Redfish iLO RESTful API. "